util/otel/pipeline.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2022 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package otel
```

Comment:
This whole file is untested, could we implement a few that makes sure the pipeline works as expected?

It could be something simple that just sends data to a mock and we assert that data that got in, eventually got out in the same format (using real processors would be out of scope here)

Reply:
example added